src/main/java/frc/robot/subsystems/CommandFactory.java

```diff
@@ -50,21 +53,35 @@ public CommandFactory(
   /*
    * Moves the robot to a position and then runs the secondary commands
    */
-  public Command moveToPositionWithDistance(Pose2d position, Distance distance, Command ... secondaryCommands) { 
-    Command goToPose = m_drivetrain.goToPose(position);
-    BooleanSupplier isClose = ()-> position.getTranslation().getDistance(m_robotPoseSupplier.get().getTranslation()) <= distance.in(Meters);
-    Command then = Commands.waitUntil(isClose).andThen(secondaryCommands);
-    Command returnCommand = Commands.parallel(goToPose,then); 
+  public Command moveToPositionWithDistance(Supplier<Pose2d> position, Distance distance, Supplier<Command> ... secondaryCommands) {
+    HashSet<Subsystem> requirements = new HashSet<>();
+    requirements.add(m_drivetrain);
+
+    Supplier<Command> gotoPoseSupplier = () -> goToPose(position.get());
+
+    Command goToPose = Commands.defer(gotoPoseSupplier,requirements);
+    BooleanSupplier isClose = ()-> position.get().getTranslation().getDistance(m_robotPoseSupplier.get().getTranslation()) <= distance.in(Meters);
+    
+    // make a new array of commands since Commands.defferedProxy wont accept an array
+    Command[] deferedSecondaryCommands = new Command[secondaryCommands.length];
+    for (int i = 0; i<secondaryCommands.length;i++) {
+      deferedSecondaryCommands[i] = Commands.deferredProxy(secondaryCommands[i]);
+    }
```

Comment:
Or could you change it to this?
```suggestion
    for (Command secondaryCommand : secondaryCommands) {
      deferedSecondaryCommands[i] = Commands.deferredProxy(secondaryCommand);
    }
```